### PR TITLE
First draft of rx vs imperative way blog post.

### PR DESCRIPTION
**Pull request by abhishekBansal**

### CODE REVIEW

File: _posts/2017-08-28-the-rx-way.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+```
+.map(MyObject::toList)
+```
+`Map` is transformation operator which allows us to convert things. Here we are converting `Map` to `List`. `toList` is a static method written in `MyObject` class. This method converts a `Map<ObjectId, MyObject>` to `List<MyObject>`. List version has object id's in object itself. `MyObject::toList` is Java 8 way of calling this method, Its named as [Method Reference](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html).
+
+```
+.flatMap(rx.Observable::from)
+```
+
+This statement allows us to receive each `MyObject` as separate event on observable stream. Before this we were having whole List or Map as single event on observable stream. To perform filtering and API calls on each of these separately we need to have them as independent events on stream.
```
Comment (aksswami) [inline _posts/2017-08-28-the-rx-way.md:101]: Maybe you can explain here how stream of stream is working and how we are again combining them?

Comment (abhishekBansal) [reply _posts/2017-08-28-the-rx-way.md:101]: My primary objective of this post was to demonstrate reactive programming abilities. I do not want to dive much into how Rx works but what its capable of doing. This is in order to keep things simple for reader. Do you think this explanation is absolutely needed for fulfilling primary objective?

Comment (anujmiddha) [reply _posts/2017-08-28-the-rx-way.md:101]: I think this should be enough to spark the curiosity of the casual reader, and that is what we want to do. An in depth discussion of streams can warrant its own post.

File: _posts/2017-08-28-the-rx-way.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+```
+.subscribeOn(Schedules.io())
+```
+`subscribeOn` is an Rx operator which directs RxJava to start processing on scheduler thread (a background thread). Just this line lets you switch your processing to a background thread. How cool is that!
+
+```
+.map(MyObject::toList)
+```
+`Map` is transformation operator which allows us to convert things. Here we are converting `Map` to `List`. `toList` is a static method written in `MyObject` class. This method converts a `Map<ObjectId, MyObject>` to `List<MyObject>`. List version has object id's in object itself. `MyObject::toList` is Java 8 way of calling this method, Its named as [Method Reference](https://docs.oracle.com/javase/tutorial/java/javaOO/methodreferences.html).
```
Comment (anujmiddha) [inline _posts/2017-08-28-the-rx-way.md]: Can we give an example of the transformed JSON?

**Review by anujmiddha — APPROVED** [review]

**Pull request merged**